Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp

```diff
@@ -1048,6 +1048,77 @@ static gboolean webkitWebViewAccumulatorObjectHandled(GSignalInvocationHint*, GV
     return !object;
 }
 
+#if PLATFORM(GTK) && USE(GTK4)
+static GdkEvent* gValueGetEvent(const GValue* value)
+{
+    g_return_val_if_fail(G_VALUE_HOLDS(value, GDK_TYPE_EVENT), NULL);
+
+    return reinterpret_cast<GdkEvent*>(value->data[0].v_pointer);
+}
+
+typedef gboolean (*ContextMenuCallback) (gpointer, WebKitContextMenu*, GdkEvent*, WebKitHitTestResult*, gpointer);
+
+static void webkitWebViewContextMenuMarshal(GClosure* closure, GValue* returnValue, guint nParams, const GValue* params, gpointer, gpointer marshalData)
+{
+    GCClosure* cc = (GCClosure *) closure;
+    gpointer data1, data2;
+
+    g_return_if_fail(returnValue);
+    g_return_if_fail(nParams == 4);
+
+    if (G_CCLOSURE_SWAP_DATA(closure)) {
+        data1 = closure->data;
+        data2 = g_value_peek_pointer(&params[0]);
+    } else {
+        data1 = g_value_peek_pointer(&params[0]);
+        data2 = closure->data;
+    }
+
+    auto* menu = WEBKIT_CONTEXT_MENU(g_value_get_object(&params[1]));
+    auto* event = gValueGetEvent(&params[2]);
+    auto* result = WEBKIT_HIT_TEST_RESULT(g_value_get_object(&params[3]));
+
+    ContextMenuCallback callback = (ContextMenuCallback) (marshalData ? marshalData : cc->callback);
+    gboolean ret = callback(data1, menu, event, result, data2);
+
+    g_value_set_boolean(returnValue, ret);
```

Comment:
Style nit: we generally use fewer blank lines:

```suggestion
    gboolean ret = callback(data1, menu, event, result, data2);
    g_value_set_boolean(returnValue, ret);
```